tests/integration/delete_user.t

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/perl -w
+
+use ProductOpener::PerlStandards;
+
+use Test::More;
+use ProductOpener::APITest qw/:all/;
+use ProductOpener::Test qw/:all/;
+use ProductOpener::TestDefaults qw/:all/;
+use ProductOpener::Users qw/:all/;
+use ProductOpener::Producers qw/:all/;
+
+remove_all_users();
+wait_application_ready();
+
+#new common user agent
+my $ua = new_client();
+my %create_client_args = (%default_user_form, (email => 'bob@test.com'));
+create_user($ua, \%create_client_args);
+
+#new admin user agent
+my $admin = new_client();
+create_user($admin, \%admin_user_form);
+
+#common ua add a new product then delete the account while being still logged in
+edit_product($ua, \%default_product);
+
+my @words = (
+	"User is being deleted. This may take a few minutes.",
+	"See you soon!",
+	"Invalid user.",
+	"Unknown user.",
+	"Incorrect user name or password.",
+	"/editor/anonymous"
+);
+my $url_userid = construct_test_url("/cgi/user.pl?type=edit&userid=tests", "world");
+my $url_delete = construct_test_url("/cgi/user.pl", "world");
+my $response_edit = $ua->get($url_userid);
+
+my %delete_form = (
+	name => 'Test',
+	email => 'bob@test.com',
+	password => '',
+	confirm_password => '',
+	delete => 'on',
+	action => 'process',
+	type => 'edit',
+	userid => 'tests'
+);
+
+#checking if the delete button exist
+like($response_edit->content, qr/Delete the user/, "the delete button does exist");
+
+#deleting the account
+my $before_delete_ts = time();
+my $response_delete = $ua->post($url_delete, \%delete_form);
+#checking if we are redirected to the account deleted page
+like($response_delete->content, qr/User is being deleted\. This may take a few minutes\./, "the account was deleted");
+
+#waiting the deletion task to be done
+my $max_time = 400;
+my $jobs_ref = get_minion_jobs("delete_user_task", $before_delete_ts, $max_time);
+
+#checking if there are remaining jobs
+my $count = 0;
+while (my $job = $jobs_ref->next) {
+	$count++;
+}
+is($count, 0, "waiting job is completed");
```

Comment:
This is a strange pattern and I don't understand how it works…

Taking into account my suggestion above, I would instead check that there was only one job, and it's finished.

```suggestion
my $jobs_ref = get_minion_jobs("delete_user_task", $before_delete_ts, $max_time);

is(scalar @{$jobs_ref}, 1, "One delete_user_task was triggered");
is(@{$jobs_ref}[0]->state, "finished", "delete_user_task finished without errors");
```

Reply:
I did it this way so that I could check if there was any remaining task to be done because I could get the "length" of the jobs because it was not a list. But if get_minion_jobs returns a list of job doing it this way is cleaner 😮